### ISSUE TITLE
Support Hooks

### ISSUE THREAD

**Issue by x6j8x** [aws-cloudformation/cloudformation-cli-go-plugin#202]
tldr;

Support writing Hooks in Go - https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks.html

**Comment by anderseknert** [comment]
Yeah, really confusing how `cfn init` provides `go` as an alternative for the `hook` option, and then goes on to render code that can't be used for that purpose.

**Comment by x6j8x** [comment]
Hi, 

as the AWS encryption sdk is now available for Go, is there a timeline when this "high priority" feature will be available for the Go plugin?

https://aws.amazon.com/about-aws/whats-new/2025/01/aws-encryption-sdk-go-available/
